Pod/Illuminator/IlluminatorElement.swift

```diff
@@ -0,0 +1,425 @@
+            return acc.reduce(appString) { (parentStr: String, elem) in elem.toXCUIElementStringWith(parentStr) }
+            
+        }
+        
+        guard elementType != XCUIElementType.Application else { return finish(appString) }
+        guard let parent = parent else {
+            print("Warning about toXCUIElementStringHelper not knowing it's done: \(toString())")
+            return finish(appString)
+        }
+        return parent.toXCUIElementStringHelper([self] + acc, appString: appString)
+    }
+    
+    // given the parent element, get the string that comes from this element
+    func toXCUIElementStringWith(parent: String) -> String {
+        guard elementType != XCUIElementType.Other || index != nil else { return parent }
+        guard !isMainWindow else { return parent }
+        
+        let prefix = parent + ".\(IlluminatorElement.elementStringFromElementType(elementType))"
+        
+        // fall back on numeric index
+        guard let idx = index else {
+            if let nidx = numericIndex {
+                return "\(prefix).elementAtIndex(\(nidx))"
+            } else {
+                return "\(prefix).elementAtIndex(-1)"
+            }
+        }
+
+        return "\(prefix)[\"\(idx)\"]"
+    }
+    
+    // given the app string, work out the full string representing this element
+    func toXCUIElementString(appString: String) -> String? {
+        return toXCUIElementStringHelper([], appString: appString)
+    }
+    
+    // get a dictionary of label -> element for the children
+    func childrenMatchingType(elementType: XCUIElementType) -> [IlluminatorElement] {
+        
+        return children.reduce([IlluminatorElement]()) { arr, elem in
+            guard elem.elementType == elementType else { return arr }
+            var ret = arr
+            ret.append(elem)
+            return ret
+        }
+    }
+    // get a dictionary of label -> element for the children
+    func childrenMatchingTypeDict(elementType: XCUIElementType) -> [String: IlluminatorElement] {
+        
+        return children.reduce([String: IlluminatorElement]()) { dict, elem in
+            guard let idx = elem.index else { return dict }
+            guard elem.elementType == elementType else { return dict }
+            var ret = dict
+            ret[idx] = elem
+            return ret
+        }
+    }
+    
+    func reduce<T>(initialResult: T, nextPartialResult: (T, IlluminatorElement) throws -> T) rethrows -> T {
+        return try children.reduce(nextPartialResult(initialResult, self)) { (acc, nextElem) in
+            return try nextElem.reduce(acc, nextPartialResult: nextPartialResult)
+        }
+    }
+    
+    
+    static func elementTypeFromString(_ description: String) -> XCUIElementType {
+        guard let val = theXCUIElementTypeOfDebugString[description] else { return .Other }
+        return val
+    }
+    
+    static func stringFromElementType(_ theType: XCUIElementType) -> String {
```

Comment:
I don't think this is possible since `XCUIElementType` appears to be an enum inside `XCUIElement`